eclair-core/src/test/scala/fr/acinq/eclair/payment/Bolt12InvoiceSpec.scala

```diff
@@ -58,219 +58,123 @@ class Bolt12InvoiceSpec extends AnyFunSuite {
     val (nodeKey, payerKey, chain) = (randomKey(), randomKey(), randomBytes32())
     val offer = Offer(Some(10000 msat), "test offer", nodeKey.publicKey, Features.empty, chain)
     val request = InvoiceRequest(offer, 11000 msat, 1, Features.empty, payerKey, chain)
-    val invoice = Bolt12Invoice(offer, request, randomBytes32(), nodeKey, CltvExpiryDelta(20), Features.empty, Seq(createPaymentBlindedRoute(nodeKey.publicKey)))
-    assert(invoice.isValidFor(offer, request))
-    assert(invoice.checkSignature())
-    assert(!invoice.checkRefundSignature())
+    val invoice = Bolt12Invoice(request, randomBytes32(), nodeKey, 300 seconds, Features.empty, Seq(createPaymentBlindedRoute(nodeKey.publicKey)))
+    assert(invoice.isValidFor(request))
     assert(Bolt12Invoice.fromString(invoice.toString).get.toString == invoice.toString)
     // changing signature makes check fail
     val withInvalidSignature = Bolt12Invoice(TlvStream(invoice.records.records.map { case Signature(_) => Signature(randomBytes64()) case x => x }, invoice.records.unknown))
-    assert(!withInvalidSignature.checkSignature())
-    assert(!withInvalidSignature.isValidFor(offer, request))
-    assert(!withInvalidSignature.checkRefundSignature())
+    assert(!withInvalidSignature.isValidFor(request))
     // changing fields makes the signature invalid
     val withModifiedUnknownTlv = Bolt12Invoice(invoice.records.copy(unknown = Seq(GenericTlv(UInt64(7), hex"ade4"))))
-    assert(!withModifiedUnknownTlv.checkSignature())
-    assert(!withModifiedUnknownTlv.isValidFor(offer, request))
-    assert(!withModifiedUnknownTlv.checkRefundSignature())
-    val withModifiedAmount = Bolt12Invoice(TlvStream(invoice.records.records.map { case Amount(amount) => Amount(amount + 100.msat) case x => x }, invoice.records.unknown))
-    assert(!withModifiedAmount.checkSignature())
-    assert(!withModifiedAmount.isValidFor(offer, request))
-    assert(!withModifiedAmount.checkRefundSignature())
+    assert(!withModifiedUnknownTlv.isValidFor(request))
+    val withModifiedAmount = Bolt12Invoice(TlvStream(invoice.records.records.map { case OfferAmount(amount) => OfferAmount(amount + 100.msat) case x => x }, invoice.records.unknown))
+    assert(!withModifiedAmount.isValidFor(request))
+  }
+
+  test("check invoice signature with unknown field from invoice request") {
+    val (nodeKey, payerKey, chain) = (randomKey(), randomKey(), randomBytes32())
+    val offer = Offer(Some(10000 msat), "test offer", nodeKey.publicKey, Features.empty, chain)
+    val basicRequest = InvoiceRequest(offer, 11000 msat, 1, Features.empty, payerKey, chain)
+    val requestWithUnknownTlv = basicRequest.copy(records = TlvStream(basicRequest.records.records, Seq(GenericTlv(UInt64(87), hex"0404"))))
+    val invoice = Bolt12Invoice(requestWithUnknownTlv, randomBytes32(), nodeKey, 300 seconds, Features.empty, Seq(createPaymentBlindedRoute(nodeKey.publicKey)))
```

Comment:
You should verify that the invoice contains the unknown tlv, it makes the test clearer.